string/string.go

```diff
@@ -0,0 +1,155 @@
+package string
+
+import (
+	"strings"
+)
+
+// ToLower converts a string to lowercase.
+func ToLower(s string) string {
+	return strings.ToLower(s)
+}
+
+// ToUpper converts a string to uppercase.
+func ToUpper(s string) string {
+	return strings.ToUpper(s)
+}
+
+// Concat concatenates two strings.
+func Concat(s1, s2 string) string {
+	return s1 + s2
+}
+
+// Reverse reverses a string.
+func Reverse(s string) string {
+	// Convert the string to a rune slice to handle Unicode characters correctly
+	runes := []rune(s)
+	length := len(runes)
+	reversed := make([]rune, length)
+
+	for i, j := length-1, 0; i >= 0; i, j = i-1, j+1 {
+		reversed[j] = runes[i]
+	}
+
+	return string(reversed)
+}
+
+// Contains checks if a string contains a substring.
+func Contains(s, substring string) bool {
+	return strings.Contains(s, substring)
+}
+
+// CountSubstring counts the number of occurrences of a substring in a string.
+func CountSubstring(s, substring string) int {
+	return strings.Count(s, substring)
+}
+
+// ReplaceAll replaces all occurrences of a substring with another string.
+func ReplaceAll(s, old, new string) string {
+	return strings.ReplaceAll(s, old, new)
+}
+
+// TrimWhitespace trims leading and trailing whitespace from a string.
+func TrimWhitespace(s string) string {
+	return strings.TrimSpace(s)
+}
+
+// Split splits a string into substrings based on a delimiter and returns a slice of substrings.
+func Split(s, delimiter string) []string {
+	return strings.Split(s, delimiter)
+}
+
+// Join concatenates elements of a string slice into a single string using a separator.
+func Join(strs []string, separator string) string {
```

Comment:
May be this can the function signature can allow the variable argument list.
func Join( separator string,s ... string) {
...code
}